src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/OptionsRuntimeTests.cs

```diff
@@ -399,6 +399,26 @@ public void TestCustomGeneratedAttributes()
 
             Assert.Equal(results.Count(), generatorResult.Failures.Count());
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
+        public void TestGeneratedRangeAttributeThreadSafety()
+        {
+            OptionsWithTimeSpanRangeAttribute options = new OptionsWithTimeSpanRangeAttribute() { Name = "T1", Period = TimeSpan.FromHours(1) };
+            TimeSpanRangeAttributeValidator validator = new TimeSpanRangeAttributeValidator();
+
+            Task[] tasks = new Task[8];
+            for (int i = 0; i < tasks.Length; i++)
+            {
+                tasks[i] = Task.Run(() =>
+                {
+                    Task.Delay(10).Wait();
```

Comment:
What is the purpose of the Delay(10).Wait here?  Is it to try to get all of the tasks to wake up at the same time?

It'd be better to use a Barrier and also to use new threads rather than thread pool threads that might be tied up running other tests, e.g.
```C#
var barrier = new Barrier(8);
Task.WaitAll(
    (from i in Enumerable.Range(0, barrier.ParticipantCount)
     select Task.Factory.StartNew(() =>
     {
         barrier.SignalAndWait();
         ValidateOptionsResult result = validator.Validate("T1", options);
         Assert.True(result.Succeeded);
     }, TaskCreationOptions.LongRunning)).ToArray());
```

Reply:
Done. thanks for the suggestion.